src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g

```diff
@@ -1224,15 +1224,26 @@ resources
     ;
 
 
-resource
-    : IDENT
-      | modifiers typeSpec[true] IDENT resource_assign
-      {#resource = #([RESOURCE, "RESOURCE"], #resource);}
+tryResourceDeclarator![AST mods, AST t]
```

Comment:
please move this after `resource:` declaration.
The same for `tryResourceDeclaration!:`.
Details of implementation should be right below first usage.

Reply:
done @romani